hydra/_internal/hydra.py

```diff
@@ -204,7 +207,9 @@ def show_cfg(
             if package is not None:
                 print(f"# @package {package}")
             if resolve:
-                OmegaConf.resolve(ret)
+                with read_write(cfg.hydra):
+                    OmegaConf.resolve(ret)
```

Comment:
This seems unrelated. What is the motivation for this? (Is there a missing test?)

Reply:
The `HydraConfig.set_config` method sets a `readonly` flag on `cfg.hydra`, so `OmegaConf.resolve` fails unless the flag is turned off.

Reply:
We could unset the readonly flag in show_cfg and show_help.

Reply:
Done in https://github.com/facebookresearch/hydra/pull/1696/commits/57452d5b9e71485cb3e92d6525f2d71a2cb3abc5. I only needed to unset the flag in `show_cfg`. In `show_help`, a readonly flag is no problem because we are using `OmegaConf.to_yaml(..., resolve=...)` instead of `OmegaConf.resolve(...)`.